<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1280</width>
    <height>720</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="s_mainWindowGridLayout">
    <item row="0" column="0">
     <spacer name="horizontalSpacer">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
      <property name="sizeHint" stdset="0">
       <size>
        <width>40</width>
        <height>20</height>
       </size>
      </property>
     </spacer>
    </item>
    <item row="0" column="1">
     <widget class="QToolBox" name="toolBox">
      <property name="maximumSize">
       <size>
        <width>310</width>
        <height>16777215</height>
       </size>
      </property>
      <property name="frameShape">
       <enum>QFrame::Box</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Sunken</enum>
      </property>
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="m_page1_flock">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>0</y>
         <width>308</width>
         <height>529</height>
        </rect>
       </property>
       <attribute name="label">
        <string>Flock</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout">
        <item>
         <widget class="QGroupBox" name="groupBox">
          <property name="title">
           <string>General</string>
          </property>
          <layout class="QGridLayout" name="gridLayout">
           <item row="0" column="1">
            <widget class="QLabel" name="label">
             <property name="text">
              <string>Flock density:</string>
             </property>
             <property name="buddy">
              <cstring>m_flockDensity</cstring>
             </property>
            </widget>
           </item>
           <item row="0" column="2">
            <widget class="QSpinBox" name="m_flockDensity">
             <property name="toolTip">
              <string>Sets the total number of Boids in the Flock</string>
             </property>
             <property name="statusTip">
              <string>Sets the total number of Boids in the Flock</string>
             </property>
             <property name="minimum">
              <number>20</number>
             </property>
             <property name="maximum">
              <number>2000</number>
             </property>
             <property name="value">
              <number>200</number>
             </property>
            </widget>
           </item>
           <item row="1" column="2">
            <widget class="QPushButton" name="m_applyFlock">
             <property name="enabled">
              <bool>false</bool>
             </property>
             <property name="toolTip">
              <string>Set the new number of boids in the flock</string>
             </property>
             <property name="statusTip">
              <string>Set the new number of boids in the flock</string>
             </property>
             <property name="text">
              <string>Apply</string>
             </property>
             <property name="flat">
              <bool>false</bool>
             </property>
            </widget>
           </item>
           <item row="1" column="1">
            <widget class="QPushButton" name="m_resetFlock">
             <property name="text">
              <string>Reset Flock</string>
             </property>
            </widget>
           </item>
           <item row="3" column="1">
            <widget class="QPushButton" name="m_addBoids">
             <property name="toolTip">
              <string>Adds 10 boids to the current flock</string>
             </property>
             <property name="statusTip">
              <string>Adds 10 boids to the current flock</string>
             </property>
             <property name="text">
              <string>+ Add Boids</string>
             </property>
            </widget>
           </item>
           <item row="3" column="2">
            <widget class="QPushButton" name="m_removeBoids">
             <property name="toolTip">
              <string>Removes 10 boids from the current flock</string>
             </property>
             <property name="statusTip">
              <string>Removes 10 boids from the current flock</string>
             </property>
             <property name="text">
              <string>- Remove Boids</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="groupBox_2">
          <property name="title">
           <string>Appearence</string>
          </property>
          <layout class="QGridLayout" name="gridLayout_2">
           <item row="0" column="0" colspan="2">
            <widget class="QPushButton" name="m_changeFlockColour">
             <property name="toolTip">
              <string>Change the flock colour</string>
             </property>
             <property name="statusTip">
              <string>Change the flock colour</string>
             </property>
             <property name="text">
              <string>Change Flock Colour</string>
             </property>
            </widget>
           </item>
           <item row="5" column="0">
            <widget class="QLabel" name="label_2">
             <property name="text">
              <string>Set Boid Size :</string>
             </property>
             <property name="buddy">
              <cstring>m_changeBoidSize</cstring>
             </property>
            </widget>
           </item>
           <item row="5" column="1">
            <widget class="QDoubleSpinBox" name="m_changeBoidSize">
             <property name="toolTip">
              <string>Changes the size of each boid in the flock</string>
             </property>
             <property name="statusTip">
              <string>Changes the size of each boid in the flock</string>
             </property>
             <property name="minimum">
              <double>0.500000000000000</double>
             </property>
             <property name="maximum">
              <double>10.000000000000000</double>
             </property>
             <property name="singleStep">
              <double>0.100000000000000</double>
             </property>
             <property name="value">
              <double>1.000000000000000</double>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QCheckBox" name="m_renderFlockWireframe">
          <property name="toolTip">
           <string>Renders the boids in wireframe mode if checked</string>
          </property>
          <property name="statusTip">
           <string>Renders the boids in wireframe mode if checked</string>
          </property>
          <property name="text">
           <string>Render Wireframe</string>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="verticalSpacer">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="m_page2_obstacle">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>0</y>
         <width>308</width>
         <height>529</height>
        </rect>
       </property>
       <attribute name="label">
        <string>Obstacle</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_2">
        <item>
         <widget class="QGroupBox" name="groupBox_3">
          <property name="title">
           <string>General</string>
          </property>
          <layout class="QGridLayout" name="gridLayout_3">
           <item row="0" column="1">
            <widget class="QLabel" name="label_6">
             <property name="text">
              <string>Position:</string>
             </property>
            </widget>
           </item>
           <item row="1" column="1">
            <widget class="QDoubleSpinBox" name="m_obstaclePosX">
             <property name="minimum">
              <double>-99.000000000000000</double>
             </property>
             <property name="value">
              <double>12.000000000000000</double>
             </property>
            </widget>
           </item>
           <item row="1" column="2">
            <widget class="QDoubleSpinBox" name="m_obstaclePosY">
             <property name="minimum">
              <double>-99.000000000000000</double>
             </property>
             <property name="value">
              <double>30.000000000000000</double>
             </property>
            </widget>
           </item>
           <item row="1" column="3">
            <widget class="QDoubleSpinBox" name="m_obstaclePosZ">
             <property name="minimum">
              <double>-99.000000000000000</double>
             </property>
            </widget>
           </item>
           <item row="2" column="1">
            <widget class="QLabel" name="label_3">
             <property name="text">
              <string>Size:</string>
             </property>
             <property name="buddy">
              <cstring>m_obstacleSize</cstring>
             </property>
            </widget>
           </item>
           <item row="2" column="2" colspan="2">
            <widget class="QDoubleSpinBox" name="m_obstacleSize">
             <property name="minimum">
              <double>1.000000000000000</double>
             </property>
             <property name="maximum">
              <double>10.000000000000000</double>
             </property>
             <property name="value">
              <double>4.000000000000000</double>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="groupBox_5">
          <property name="title">
           <string>Appearence</string>
          </property>
          <layout class="QGridLayout" name="gridLayout_4">
           <item row="0" column="0">
            <widget class="QPushButton" name="m_obstacleColour">
             <property name="text">
              <string>Change Colour</string>
             </property>
            </widget>
           </item>
           <item row="1" column="0">
            <widget class="QCheckBox" name="m_obstacleWireframe">
             <property name="text">
              <string>Render Wireframe</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <spacer name="verticalSpacer_2">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="m_page3_simulation">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>0</y>
         <width>308</width>
         <height>529</height>
        </rect>
       </property>
       <attribute name="label">
        <string>Simulation</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_4">
        <item>
         <widget class="QGroupBox" name="groupBox_7">
          <property name="title">
           <string>General</string>
          </property>
          <layout class="QGridLayout" name="gridLayout_7">
           <item row="0" column="0">
            <widget class="QLabel" name="label_5">
             <property name="text">
              <string>Neighbourhood Distance :</string>
             </property>
            </widget>
           </item>
           <item row="1" column="0">
            <widget class="QDoubleSpinBox" name="m_simDistance">
             <property name="value">
              <double>20.000000000000000</double>
             </property>
            </widget>
           </item>
           <item row="4" column="0">
            <widget class="QLabel" name="label_7">
             <property name="text">
              <string>Cohesion Force :</string>
             </property>
            </widget>
           </item>
           <item row="5" column="0">
            <widget class="QDoubleSpinBox" name="m_simCohesion">
             <property name="value">
              <double>2.000000000000000</double>
             </property>
            </widget>
           </item>
           <item row="6" column="0">
            <widget class="QLabel" name="label_8">
             <property name="text">
              <string>Separation Force :</string>
             </property>
            </widget>
           </item>
           <item row="7" column="0">
            <widget class="QDoubleSpinBox" name="m_simSeparation">
             <property name="value">
              <double>9.000000000000000</double>
             </property>
            </widget>
           </item>
           <item row="8" column="0">
            <widget class="QLabel" name="label_9">
             <property name="text">
              <string>Alignment Weight :</string>
             </property>
            </widget>
           </item>
           <item row="9" column="0">
            <widget class="QDoubleSpinBox" name="m_simAlignment">
             <property name="value">
              <double>10.000000000000000</double>
             </property>
            </widget>
           </item>
           <item row="2" column="0">
            <widget class="QLabel" name="label_10">
             <property name="text">
              <string>Flock Distance</string>
             </property>
            </widget>
           </item>
           <item row="3" column="0">
            <widget class="QDoubleSpinBox" name="m_simFlockDistance">
             <property name="value">
              <double>4.000000000000000</double>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="m_page4_environment">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>0</y>
         <width>308</width>
         <height>529</height>
        </rect>
       </property>
       <attribute name="label">
        <string>Environment</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_3">
        <item>
         <widget class="QGroupBox" name="groupBox_4">
          <property name="title">
           <string>Bounding Box</string>
          </property>
          <layout class="QGridLayout" name="gridLayout_5">
           <item row="0" column="0">
            <widget class="QLabel" name="label_4">
             <property name="text">
              <string>Size :</string>
             </property>
             <property name="buddy">
              <cstring>m_bboxSize</cstring>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QDoubleSpinBox" name="m_bboxSize">
             <property name="minimum">
              <double>50.000000000000000</double>
             </property>
             <property name="maximum">
              <double>220.000000000000000</double>
             </property>
             <property name="value">
              <double>120.000000000000000</double>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="groupBox_6">
          <property name="title">
           <string>Appearence</string>
          </property>
          <layout class="QGridLayout" name="gridLayout_6">
           <item row="0" column="0">
            <widget class="QPushButton" name="m_backColour">
             <property name="text">
              <string>Background Colour</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <spacer name="verticalSpacer_3">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1280</width>
     <height>25</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <tabstops>
  <tabstop>m_flockDensity</tabstop>
  <tabstop>m_resetFlock</tabstop>
  <tabstop>m_applyFlock</tabstop>
  <tabstop>m_addBoids</tabstop>
  <tabstop>m_removeBoids</tabstop>
  <tabstop>m_changeFlockColour</tabstop>
  <tabstop>m_changeBoidSize</tabstop>
  <tabstop>m_renderFlockWireframe</tabstop>
  <tabstop>m_obstaclePosX</tabstop>
  <tabstop>m_obstaclePosY</tabstop>
  <tabstop>m_obstaclePosZ</tabstop>
  <tabstop>m_obstacleSize</tabstop>
  <tabstop>m_obstacleColour</tabstop>
  <tabstop>m_obstacleWireframe</tabstop>
  <tabstop>m_simDistance</tabstop>
  <tabstop>m_simCohesion</tabstop>
  <tabstop>m_simSeparation</tabstop>
  <tabstop>m_simAlignment</tabstop>
  <tabstop>m_bboxSize</tabstop>
  <tabstop>m_backColour</tabstop>
 </tabstops>
 <resources/>
 <connections/>
</ui>
